### ISSUE TITLE
Pylance failed to start service instance, caused by a regexp error

### ISSUE THREAD

**Issue by SurpassHR** [microsoft/pylance-release#5963]
## Environment data

-   Language Server version: v2024.5.1
-   OS and version: Windows 23H2 22631.3447
-   Python version (& distribution if applicable, e.g. Anaconda): 3.8.8

## Repro Steps

1. Create a dir with '+' in it's name.
2. Open this folder with vscode.
3. Create a .ipynb file then the python language sever instance start failed.

## Actual behavior

intellisense and autocomplete failure in jupyternb. If you change the dir name into something else without a '+', pylance works out fine.

## Logs

```
2024-06-04 15:53:33.105 [info] (Client) Pylance async client (2024.5.1) started with python extension (2024.6.0)
2024-06-04 15:53:33.725 [info] [Info  - 3:53:33 PM] (29080) Pylance language server 2024.5.1 (pyright version 1.1.360, commit 4bc8428d) starting
2024-06-04 15:53:33.725 [info] [Info  - 3:53:33 PM] (29080) Server root directory: file:///c%3A/Users/123456/.vscode/extensions/ms-python.vscode-pylance-2024.5.1/dist
2024-06-04 15:53:33.731 [info] [Info  - 3:53:33 PM] (29080) Starting service instance "c++"
2024-06-04 15:53:33.736 [info] [Info  - 3:53:33 PM] (29080) Auto-Indent enabled
2024-06-04 15:53:33.770 [info] SyntaxError: Invalid regular expression: /\$\{workspaceFolder:c++\}/g: Nothing to repeat
    at RegExp (<anonymous>)
    at c (c:\Users\123456\.vscode\extensions\ms-python.vscode-pylance-2024.5.1\dist\pyright.bundle.js:1:1155530)
    at _0x2ef442 (c:\Users\123456\.vscode\extensions\ms-python.vscode-pylance-2024.5.1\dist\server.bundle.js:1:770177)
    at _0x542e75._getPythonInformation (c:\Users\123456\.vscode\extensions\ms-python.vscode-pylance-2024.5.1\dist\server.bundle.js:1:727111)
    at _0x542e75._onDidOpenNotebookDocumentAsync (c:\Users\123456\.vscode\extensions\ms-python.vscode-pylance-2024.5.1\dist\server.bundle.js:1:733386)

2024-06-04 15:53:33.770 [info] [Error - 3:53:33 PM] (29080) Error reading settings: SyntaxError: Invalid regular expression: /\$\{workspaceFolder:c++\}/g: Nothing to repeat
2024-06-04 15:53:34.155 [info] [Info  - 3:53:34 PM] (29080) Found 2 source files
2024-06-04 15:53:34.174 [info] [Info  - 3:53:34 PM] (29080) Background analysis(1) root directory: file:///c%3A/Users/123456/.vscode/extensions/ms-python.vscode-pylance-2024.5.1/dist
2024-06-04 15:53:34.176 [info] [Info  - 3:53:34 PM] (29080) Background analysis(1) started
2024-06-04 15:53:34.554 [info] [Info  - 3:53:34 PM] (29080) Indexer background runner(2) root directory: file:///c%3A/Users/123456/.vscode/extensions/ms-python.vscode-pylance-2024.5.1/dist (index)
2024-06-04 15:53:34.554 [info] [Info  - 3:53:34 PM] (29080) Indexing(2) started
2024-06-04 15:53:35.530 [info] [Info  - 3:53:35 PM] (29080) scanned(2) 146 files over 1 exec env
2024-06-04 15:53:36.778 [info] [Info  - 3:53:36 PM] (29080) indexed(2) 98 files over 1 exec env
2024-06-04 15:53:36.836 [info] [Info  - 3:53:36 PM] (29080) Indexing finished(2).
```

**Comment by rchiodo** [comment]
Thanks for the issue. I can reproduce it:

```
2024-06-04 09:13:44.613 [error] [
  'Starting language server, Class name = d, completed in 15ms, has a falsy return value, , Return Value: undefined',
  [SyntaxError: Invalid regular expression: /\$\{workspaceFolder:test++\}/g: Nothing to repeat
  	at RegExp (<anonymous>)
  	at c (c:\Users\rchiodo\.vscode\extensions\ms-python.vscode-pylance-2024.5.104-dev.1.gddbedf500\dist\pyright.bundle.js:1:1161361)
  	at _0x5024a9.getExpandedSetting (c:\Users\rchiodo\.vscode\extensions\ms-python.vscode-pylance-2024.5.104-dev.1.gddbedf500\dist\extension.bundle.js:1:190096)
  	at _0x478089 (c:\Users\rchiodo\.vscode\extensions\ms-python.vscode-pylance-2024.5.104-dev.1.gddbedf500\dist\extension.bundle.js:1:120240)
  	at _0x1ba8f3 (c:\Users\rchiodo\.vscode\extensions\ms-python.vscode-pylance-2024.5.104-dev.1.gddbedf500\dist\extension.bundle.js:1:127538)
  	at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
  	at async _0x180447._startClient (c:\Users\rchiodo\.vscode\extensions\ms-python.vscode-pylance-2024.5.104-dev.1.gddbedf500\dist\extension.bundle.js:1:75123)
  	at async _0x180447._createNewRequest (c:\Users\rchiodo\.vscode\extensions\ms-python.vscode-pylance-2024.5.104-dev.1.gddbedf500\dist\extension.bundle.js:1:74628)
  	at async _0x180447._enqueueRequestAndWaitForRun (c:\Users\rchiodo\.vscode\extensions\ms-python.vscode-pylance-2024.5.104-dev.1.gddbedf500\dist\extension.bundle.js:1:73863)
  	at async _0x180447.startClient (c:\Users\rchiodo\.vscode\extensions\ms-python.vscode-pylance-2024.5.104-dev.1.gddbedf500\dist\extension.bundle.js:1:73326)
  	at async h.start (c:\Users\rchiodo\.vscode\extensions\ms-python.python-2024.6.0\out\client\extension.js:2:35787)]
```

For me it fails to even start the language server.

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2024.6.100, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#20246100-13-June-2024-PreRelease)
